allennlp/models/encoder_decoders/wikitables/wikitables_semantic_parser.py

```diff
@@ -86,19 +101,23 @@ def __init__(self,
                  attention_function: SimilarityFunction,
                  dropout: float = 0.0,
                  num_linking_features: int = 8,
-                 rule_namespace: str = 'rule_labels') -> None:
+                 rule_namespace: str = 'rule_labels',
+                 table_directory: str = '/wikitables/') -> None:
         super(WikiTablesSemanticParser, self).__init__(vocab)
         self._question_embedder = question_embedder
         self._encoder = encoder
         self._entity_encoder = TimeDistributed(entity_encoder)
         self._beam_search = decoder_beam_search
         self._max_decoding_steps = max_decoding_steps
         self._action_sequence_accuracy = Average()
+        self._denotation_accuracy = Average()
+        self._has_logical_form = Average()
```

Comment:
`_has_logical_form` -> `_logical_form_is_valid`?

Reply:
It's not that we're producing invalid logical forms, it's that the beam search never found a finished state before it ran out of steps.  So it would be `action_sequence_is_valid`; `has_logical_form` seems just as descriptive.